### ISSUE TITLE
Quokka not functional

### ISSUE THREAD

**Issue by ivankolev** [wallabyjs/quokka#65]
### Issue description or question
Quokka not functional. Fresh install of the plugin, Intellij IDEA Ultimate 2017.2 (I have tried with the latest EAP too), Windows 10 Pro. I have noticed the .quokka folder is empty. The IDE is run as Admin. On every IDE start it displays the prompt for the Pro version, which should not be the case.  

### Quokka.js file
any scratch file

### Quokka.js Console Output
It just shows Quokka #2, notably there's missing the node version string. I have node 6.10.0 installed and operational on this machine

### Code editor version
IntelliJ IDEA 2017.2.1 EAP
Build #IU-172.3544.6, built on July 19, 2017
JRE: 1.8.0_152-release-915-b6 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0

**Comment by ArtemGovorov** [comment]
In this case there should be something Quokka related written to the idea.log (`Help - Show Log`). Could you please check it and paste relevant log entries from Quokka?

**Comment by ivankolev** [comment]
Right, here's the stack trace taken from the log:
```
017-07-21 21:38:35,088 [  18183]  ERROR - llij.ide.plugins.PluginManager - window with id="Quokka" is already registered 
java.lang.IllegalArgumentException: window with id="Quokka" is already registered
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.a(ToolWindowManagerImpl.java:1150)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.initToolWindow(ToolWindowManagerImpl.java:465)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl$4.run(ToolWindowManagerImpl.java:451)
	at com.intellij.openapi.wm.impl.CommandProcessor.a(CommandProcessor.java:93)
	at com.intellij.openapi.wm.impl.CommandProcessor.flush(CommandProcessor.java:50)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.c(ToolWindowManagerImpl.java:573)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.access$3500(ToolWindowManagerImpl.java:91)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl$InitToolWindowsActivity.a(ToolWindowManagerImpl.java:2499)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:821)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:649)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:365)
	at com.intellij.ide.IdeEventQueue.pumpEventsForHierarchy(IdeEventQueue.java:923)
	at com.intellij.openapi.progress.util.ProgressWindow.startBlocking(ProgressWindow.java:216)
	at com.intellij.openapi.progress.util.ProgressWindow.startBlocking(ProgressWindow.java:202)
	at com.intellij.openapi.application.impl.ApplicationImpl.runProcessWithProgressSynchronously(ApplicationImpl.java:617)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcessWithProgressSynchronously(CoreProgressManager.java:414)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.runProcessWithProgressSynchronously(ProgressManagerImpl.java:117)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:332)
	at com.intellij.openapi.progress.impl.CoreProgressManager.run(CoreProgressManager.java:317)
	at com.intellij.openapi.progress.ProgressManager.run(ProgressManager.java:190)
	at com.intellij.openapi.project.impl.ProjectManagerImpl.loadAndOpenProject(ProjectManagerImpl.java:476)
	at com.intellij.ide.impl.ProjectUtil.openProject(ProjectUtil.java:191)
	at com.intellij.ide.RecentProjectsManagerImpl.doOpenProject(RecentProjectsManagerImpl.java:63)
	at com.intellij.ide.RecentProjectsManagerBase.doReopenLastProject(RecentProjectsManagerBase.java:674)
	at com.intellij.ide.RecentProjectsManagerBase$MyAppLifecycleListener.appStarting(RecentProjectsManagerBase.java:716)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.util.messages.impl.MessageBusConnectionImpl.deliverMessage(MessageBusConnectionImpl.java:117)
	at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:443)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpWaitingBuses(MessageBusImpl.java:404)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:393)
	at com.intellij.util.messages.impl.MessageBusImpl.sendMessage(MessageBusImpl.java:374)
	at com.intellij.util.messages.impl.MessageBusImpl.access$200(MessageBusImpl.java:45)
	at com.intellij.util.messages.impl.MessageBusImpl$2.invoke(MessageBusImpl.java:231)
	at com.sun.proxy.$Proxy64.appStarting(Unknown Source)
	at com.intellij.idea.IdeaApplication$IdeStarter.a(IdeaApplication.java:363)
	at com.intellij.openapi.application.TransactionGuardImpl.a(TransactionGuardImpl.java:86)
	at com.intellij.openapi.application.TransactionGuardImpl.a(TransactionGuardImpl.java:109)
	at com.intellij.openapi.application.TransactionGuardImpl.submitTransaction(TransactionGuardImpl.java:118)
	at com.intellij.openapi.application.TransactionGuard.submitTransaction(TransactionGuard.java:122)
	at com.intellij.idea.IdeaApplication$IdeStarter.main(IdeaApplication.java:361)
	at com.intellij.idea.IdeaApplication.b(IdeaApplication.java:206)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:193)
	at com.intellij.idea.IdeaApplication.run(IdeaApplication.java:206)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.idea.MainImpl$1$1.a(MainImpl.java:67)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:345)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by ArtemGovorov** [comment]
Is there anything else for Quokka in the log? I'd expect some more errors.

**Comment by ivankolev** [comment]
Nope, no errors. The only other occurrences of "Quokka" are:
```
2017-07-21 22:23:05,508 [  11494]   INFO - ij.components.ComponentManager - com.quokka.javascript.QuokkaController initialized in 62 ms
```
and 
```
INFO - llij.ide.plugins.PluginManager - Loaded bundled plugins ... Quokka (1.0.17) ...
```

I have bumped the node version to 7.10.1a and reinstalled Quokka. It still does not populate the .quokka folder on launch. 
Thanks for the prompt responses Artem, and sorry for my not so prompt one. Let me know what else I can poke around and I will do tomorrow morning (I am EST and going to bed now)

**Comment by ArtemGovorov** [comment]
Thanks for checking.

I have added some extra debug logging for Quokka plugin in IDE log. Could you please update your Quokka plugin to 1.0.18 to see what's logged now?

You can then turn the verbose logging on by opening the menu `"Help - Configure Debug Log Settings...`" and adding `Quokka` there. Right after doing it, you may restart the IDE, start Quokka, **select any option in the prompt for the Pro version** (this should cause creating global config file in the .quokka folder, or log the error), then you may open the IDE log. It should contain some additional debugging info for Quokka. Could you please paste it here?

**Comment by ivankolev** [comment]
Ok, did that and I've got this:
```
INFO - lij.diagnostic.DebugLogManager - Set DEBUG for the following categories: #Quokka
```
and then this:
```
WARN -                         Quokka - C:\USERPATH\.quokka\config.json (The system cannot find the file specified) 
java.io.FileNotFoundException: C:\USERPATH\.quokka\config.json (The system cannot find the file specified)
	at java.io.FileInputStream.$$YJP$$open0(Native Method)
	at java.io.FileInputStream.open0(FileInputStream.java)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at com.intellij.openapi.util.io.FileUtilRt.loadLines(FileUtilRt.java:611)
	at com.intellij.openapi.util.io.FileUtilRt.loadLines(FileUtilRt.java:606)
	at com.intellij.openapi.util.io.FileUtilRt.loadLines(FileUtilRt.java:596)
	at com.quokka.javascript.QuokkaController.a(Unknown Source)
	at com.quokka.javascript.QuokkaController.b(Unknown Source)
	at com.quokka.javascript.p.hyperlinkUpdate(Unknown Source)
	at com.intellij.notification.impl.ui.NotificationsUtil$1.hyperlinkUpdate(NotificationsUtil.java:155)
	at javax.swing.JEditorPane.fireHyperlinkUpdate(JEditorPane.java:342)
	at javax.swing.text.html.HTMLEditorKit$LinkController.activateLink(HTMLEditorKit.java:875)
	at javax.swing.text.html.HTMLEditorKit$LinkController.mouseClicked(HTMLEditorKit.java:674)
	at java.awt.AWTEventMulticaster.mouseClicked(AWTEventMulticaster.java:270)
	at java.awt.AWTEventMulticaster.mouseClicked(AWTEventMulticaster.java:269)
	at java.awt.AWTEventMulticaster.mouseClicked(AWTEventMulticaster.java:269)
	at java.awt.Component.processMouseEvent(Component.java:6544)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6306)
	at java.awt.Container.processEvent(Container.java:2237)
	at java.awt.Component.dispatchEventImpl(Component.java:4897)
	at java.awt.Container.dispatchEventImpl(Container.java:2295)
	at java.awt.Component.dispatchEvent(Component.java:4719)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4889)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4467)
	at java.awt.Container.dispatchEventImpl(Container.java:2281)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4719)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:764)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:737)
	at java.awt.EventQueue$4.run(EventQueue.java:735)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:734)
	at com.intellij.ide.IdeEventQueue.k(IdeEventQueue.java:821)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:645)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:365)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by ArtemGovorov** [comment]
Thanks for checking!

There was indeed an issue with the initial config file saving (fixed in the latest v1.0.19), however, it shouldn't have stopped Quokka from running properly, so I suspect there're some other issues we may need to troubleshoot.

Please update to the latest version (1.0.19) and try again. This time, after the version selection, the new file should be created in the `.quokka` folder. Then try creating a new Quokka file after it and typing something in it. If it doesn't work, please share the idea.log output for Quokka this time.

**Comment by ivankolev** [comment]
Confirmed, this latest update fixed it. Keep up the awesome work guys, thanks!

**Comment by ArtemGovorov** [comment]
Awesome, thanks for the update!

**Comment by loganpowell** [comment]
Hi! Quokka worked marvelously when I was using Atom, but now that I switched over to the latest version of Intellij IDEA (2018.1.5) I'm struggling to get it working. I have  a Pro license, but am getting this error:

```
Connection refused: connect
java.net.ConnectException: Connection refused: connect
	at java.net.TwoStacksPlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.Socket.connect(Socket.java:589)
	at a.a.a.b.run(Unknown Source)
	at com.quokka.javascript.session.q.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by ArtemGovorov** [comment]
@loganpowell Could you please provide some more details:
- your OS version,
- your system node.js version,
- all Quokka related entries from your idea.log (`Help - Show Log` menu) (written after you start Quokka and get the exception).

**Comment by loganpowell** [comment]
@ArtemGovorov sorry for the delayed response:

> OS version
Windows 10 (v 1803)
> System Node version
8.9.4
> Quokka related entries from idea.log
```
ERROR -                         Quokka - Connection refused: connect 
java.net.ConnectException: Connection refused: connect
	at java.net.TwoStacksPlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.Socket.connect(Socket.java:589)
	at a.a.a.b.run(Unknown Source)
	at com.quokka.javascript.session.q.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)
2018-06-18 09:06:48,749 [4291201]  ERROR -                         Quokka - IntelliJ IDEA 2018.1.5  Build #IU-181.5281.24 
2018-06-18 09:06:48,749 [4291201]  ERROR -                         Quokka - JDK: 1.8.0_152-release 
2018-06-18 09:06:48,749 [4291201]  ERROR -                         Quokka - VM: OpenJDK 64-Bit Server VM 
2018-06-18 09:06:48,749 [4291201]  ERROR -                         Quokka - Vendor: JetBrains s.r.o 
2018-06-18 09:06:48,749 [4291201]  ERROR -                         Quokka - OS: Windows 10 
2018-06-18 09:06:48,749 [4291201]  ERROR -                         Quokka - Last Action: Quokka.RunQuokkaOnFile 
2018-06-18 09:07:26,803 [4329255]  ERROR -                         Quokka - Connection timed out: connect 
java.net.ConnectException: Connection timed out: connect
	at java.net.TwoStacksPlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.Socket.connect(Socket.java:589)
	at a.a.a.b.run(Unknown Source)
	at com.quokka.javascript.session.q.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)
2018-06-18 09:07:26,804 [4329256]  ERROR -                         Quokka - IntelliJ IDEA 2018.1.5  Build #IU-181.5281.24 
2018-06-18 09:07:26,804 [4329256]  ERROR -                         Quokka - JDK: 1.8.0_152-release 
2018-06-18 09:07:26,804 [4329256]  ERROR -                         Quokka - VM: OpenJDK 64-Bit Server VM 
2018-06-18 09:07:26,804 [4329256]  ERROR -                         Quokka - Vendor: JetBrains s.r.o 
2018-06-18 09:07:26,804 [4329256]  ERROR -                         Quokka - OS: Windows 10 
2018-06-18 09:07:26,804 [4329256]  ERROR -                         Quokka - Last Action: Quokka.RunQuokkaOnFile 
2018-06-18 09:10:57,765 [4540217]   INFO - ide.actions.ShowFilePathAction - 
Exit code 1 
```

**Comment by ArtemGovorov** [comment]
@loganpowell Could you please 
- open `Help - Debug Log Settings` menu,
- add `Quokka` line there,
- run Quokka and share Quokka related entries from your idea.log (there should be some more/detailed entires now).

**Comment by loganpowell** [comment]
I've followed your instructions... Here is the log data:

```
ERROR -                         Quokka - Connection refused: connect 
java.net.ConnectException: Connection refused: connect
	at java.net.TwoStacksPlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.Socket.connect(Socket.java:589)
	at a.a.a.b.run(Unknown Source)
	at com.quokka.javascript.session.q.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)
2018-06-19 09:53:41,818 [ 488342]  ERROR -                         Quokka - IntelliJ IDEA 2018.1.5  Build #IU-181.5281.24 
2018-06-19 09:53:41,818 [ 488342]  ERROR -                         Quokka - JDK: 1.8.0_152-release 
2018-06-19 09:53:41,818 [ 488342]  ERROR -                         Quokka - VM: OpenJDK 64-Bit Server VM 
2018-06-19 09:53:41,818 [ 488342]  ERROR -                         Quokka - Vendor: JetBrains s.r.o 
2018-06-19 09:53:41,818 [ 488342]  ERROR -                         Quokka - OS: Windows 10 
2018-06-19 09:54:23,483 [ 530007]  ERROR -                         Quokka - Connection timed out: connect 
java.net.ConnectException: Connection timed out: connect
	at java.net.TwoStacksPlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.Socket.connect(Socket.java:589)
	at a.a.a.b.run(Unknown Source)
	at com.quokka.javascript.session.q.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)
2018-06-19 09:54:23,486 [ 530010]  ERROR -                         Quokka - IntelliJ IDEA 2018.1.5  Build #IU-181.5281.24 
2018-06-19 09:54:23,486 [ 530010]  ERROR -                         Quokka - JDK: 1.8.0_152-release 
2018-06-19 09:54:23,487 [ 530011]  ERROR -                         Quokka - VM: OpenJDK 64-Bit Server VM 
2018-06-19 09:54:23,487 [ 530011]  ERROR -                         Quokka - Vendor: JetBrains s.r.o 
2018-06-19 09:54:23,487 [ 530011]  ERROR -                         Quokka - OS: Windows 10 
2018-06-19 09:54:39,394 [ 545918]   INFO - lij.diagnostic.DebugLogManager - Set DEBUG for the following categories: Quokka 
2018-06-19 09:54:49,961 [ 556485]   INFO - lij.diagnostic.DebugLogManager - Set DEBUG for the following categories: Quokka 
2018-06-19 09:55:01,874 [ 568398]  DEBUG -                         Quokka - Creating new session 
2018-06-19 09:55:01,880 [ 568404]  DEBUG -                         Quokka - Started core process 
2018-06-19 09:55:02,391 [ 568915]  DEBUG -                         Quokka - Core started, connecting to it 
2018-06-19 09:55:16,030 [ 582554]  DEBUG -                         Quokka - Restarting/reusing existing session 
2018-06-19 09:55:16,055 [ 582579]  DEBUG -                         Quokka - Started core process 
2018-06-19 09:55:16,543 [ 583067]  DEBUG -                         Quokka - Core started, connecting to it 
2018-06-19 09:55:24,398 [ 590922]  ERROR -                         Quokka - Connection refused: connect 
java.net.ConnectException: Connection refused: connect
	at java.net.TwoStacksPlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.Socket.connect(Socket.java:589)
	at a.a.a.b.run(Unknown Source)
	at com.quokka.javascript.session.q.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)
2018-06-19 09:55:24,400 [ 590924]  ERROR -                         Quokka - IntelliJ IDEA 2018.1.5  Build #IU-181.5281.24 
2018-06-19 09:55:24,400 [ 590924]  ERROR -                         Quokka - JDK: 1.8.0_152-release 
2018-06-19 09:55:24,401 [ 590925]  ERROR -                         Quokka - VM: OpenJDK 64-Bit Server VM 
2018-06-19 09:55:24,401 [ 590925]  ERROR -                         Quokka - Vendor: JetBrains s.r.o 
2018-06-19 09:55:24,401 [ 590925]  ERROR -                         Quokka - OS: Windows 10 
2018-06-19 09:55:58,549 [ 625073]  ERROR -                         Quokka - Connection timed out: connect 
java.net.ConnectException: Connection timed out: connect
	at java.net.TwoStacksPlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.Socket.connect(Socket.java:589)
	at a.a.a.b.run(Unknown Source)
	at com.quokka.javascript.session.q.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)
2018-06-19 09:55:58,551 [ 625075]  ERROR -                         Quokka - IntelliJ IDEA 2018.1.5  Build #IU-181.5281.24 
2018-06-19 09:55:58,551 [ 625075]  ERROR -                         Quokka - JDK: 1.8.0_152-release 
2018-06-19 09:55:58,551 [ 625075]  ERROR -                         Quokka - VM: OpenJDK 64-Bit Server VM 
2018-06-19 09:55:58,552 [ 625076]  ERROR -                         Quokka - Vendor: JetBrains s.r.o 
2018-06-19 09:55:58,552 [ 625076]  ERROR -                         Quokka - OS: Windows 10 
2018-06-19 10:09:13,495 [1420019]   INFO - ide.actions.ShowFilePathAction - 
Exit code 1 

```

**Comment by ArtemGovorov** [comment]
Thanks, it looks like everything is going ok from Quokka side of things, but something else is blocking a connection from IntelliJ IDEA to the local node server that Quokka spins up. My guess would be that it is some network filtering software, maybe firewall?

**Comment by loganpowell** [comment]
I just did a fresh install of Intellij and Quokka and it's working! It must have been something I did. Sorry for the trouble.

**Comment by OliverMax** [comment]
windows 11 x64, wsl2

java.net.ConnectException: Connection refused: connect
	at java.base/sun.nio.ch.Net.connect0(Native Method)
	at java.base/sun.nio.ch.Net.connect(Net.java:579)
	at java.base/sun.nio.ch.Net.connect(Net.java:568)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:593)
	at java.base/java.net.Socket.connect(Socket.java:633)
	at org.java_websocket.client.WebSocketClient.run(WebSocketClient.java:473)
	at com.quokka.javascript.session.QuokkaSession.O(Unknown Source)
	at java.base/java.lang.Thread.run(Thread.java:833)

**Comment by smcenlly** [comment]
@OliverMax - this has been addressed, please see #896.
